### ISSUE TITLE
EarTrumpet Tray Icon disappears / NIM_MODIFY Failed

### ISSUE THREAD

**Issue by schittli** [File-New-Project/EarTrumpet#389]
Good afternoon

**Thanks a lot for this great Tool!**, it enhances the usability very much and I hope to sometimes get an Non-M$-Store setup (e.g. to add it to Chocolatey :-) )

### Note: 
It's possible that _this_ issue relates to another Issue: #314 

### The Issue:
After about 1 Day, the EarTrumpet Tray Icon often disappears while EarTrumper.exe is still running.
Here, I have a Screenshot to show the situation: [EarTrumpet: Tray Icon disapeared](https://pasteboard.co/IkrJmra.png)

### The Logfile:
I have added the Logfile: https://justpaste.it/TomsEarTrumpetLog1

**The most important lines probably are:**

```
14:44:47.586 UI TaskbarIconSource Changed: C:\Windows\System32\SndVolSSO.dll,120:hc=False dpi=115 theme=False legacy=True
14:44:49.536 UI TaskbarIconSource Changed: C:\Windows\System32\SndVolSSO.dll,123:hc=False dpi=115 theme=False legacy=True
14:46:47.059 UI ShellNotifyIcon Update NIM_MODIFY Failed: 2147500037

```

I hope that the _NIM_MODIFY Failed_ error helps a lot...

Thanks a lot!, kind regards,
Thomas

**Comment by davuxcom** [comment]
Thanks, this is definitely useful. It proves to some extent what's going on.  I've added a change to cause a crash when this happens, and that'll let us assess whether or not it's a consequence of leaking handles or something else.

**Comment by MahdiGhiasi** [comment]
I've been seeing the same issue as well in the past few weeks.

(I was not sure how to find the Logfile when the icon is missing, and therefore I can't open EarTrumpet settings)

**Comment by schittli** [comment]
`I was not sure how to find the Logfile when the icon is missing, and therefore I can't open EarTrumpet settings`

Yes, this was tricky: I kept the EarTrumpet Settings » About Dialog open and waited until the Tray Icon disapeared :-)

**Comment by davuxcom** [comment]
I had the same question at first. :p Another option would be to use the assigned hotkey to open settings.  

The 'log file' isn't generated until you ask for it, so there isn't somewhere on disk to discover it.

**Comment by Oregoncat** [comment]
**I've been having the same problem lately.** The icon simply disappears from the tray, but it's still running and I can use the hotkey to open it. I can also end the task then restart it and the icon returns. Only to disappear again later.

**Comment by davuxcom** [comment]
This bug should have been fixed in 2.1.4.0 by this commit: 91da1e5467a47ff2389b50f9dd56b1e626b54480

**Comment by schittli** [comment]
#### Thank you very much, @davuxcom for your support!
Unfortunately, the issue still exists, but 

#### now I'm fortunately able to easily reproduce the disappearance of the Tray icon:
1. Start EarTrumpet
2. Kill explorer.exe
3. Start explorer.exe » The icon is not displayed again

#### Code Example for the TaskbarCreated Message
I am _very sorry_ that I can not send you a Pull Request because I'm very busy on another Non-Profit project in my spare time.

But I have googled for _Tray Icon disappears after explorer.exe crash_ and it looks like that the Windows Message `TaskbarCreated` can be used to re-assign the Tray Icon. 
I guess that this C# Code shows a useful and good designed example:

- C# Code: [WinForms NotifyIcon](http://www.dotnetframework.org/default.aspx/DotNET/DotNET/8@0/untmp/whidbey/REDBITS/ndp/fx/src/WinForms/Managed/System/WinForms/NotifyIcon@cs/1/NotifyIcon@cs)
- Searching for `TaskbarCreated` and `UpdateIcon` shows the concept, I guess that EarTrumpet already has most of the required Code :-)

Thanks a lot, kind regards,
Thomas

**Comment by riverar** [comment]
Hi @schittli are you running 2.1.4.0? It's only rolling out to 25% of users right now, we're cranking that up now.

**Comment by schittli** [comment]
@riverar Thank you for your question!, **you're right, I've still 2.1.3.0 installed :-(.** 

I tried to install it from Microsoft store two times (I have to manually download the AppxBundle and BlockMap files and install the Software offline) but I've got 2.1.3.0. 
Of course, I've expected to get the latest release and have not validated it.

**As soon as I find some time to try to install it the 3rd time, I'll validate the Download before I will test the Bugfix again.**

I'm very sorry for the troubles.
